src/apple/config/raw.rs

```diff
@@ -51,6 +51,7 @@ pub struct Raw {
     pub ios_features: Option<Vec<String>>,
     pub macos_no_default_features: Option<bool>,
     pub macos_features: Option<Vec<String>>,
+    pub use_legacy_build_system: bool,
```

Comment:
I'm pretty sure that this not being wrapped in an `Option` will result in it unconditionally being written to mobile.toml, when ideally we want it to be unspecified by default (defaulting to `true` when upgrading to non-raw `Config`)